packages/react-native/React/DevSupport/RCTPausedInDebuggerOverlayController.mm

```diff
@@ -65,82 +68,149 @@ - (void)viewDidLoad
+    [stackView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor constant:-20],
+    [stackView.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor constant:-20],
+    [stackView.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor],
   ]];
+#else
+  [self setView:stackView];
+#endif 
 
+#if !TARGET_OS_OSX // [macOS]
   stackView.semanticContentAttribute = UISemanticContentAttributeForceLeftToRight;
+#else // [macOS
+  stackView.userInterfaceLayoutDirection = NSUserInterfaceLayoutDirectionLeftToRight;
+#endif // macOS]
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)handleResume:(UITapGestureRecognizer *)recognizer
 {
   self.onResume();
 }
-#endif // [macOS]
+#else // [macOS
+- (void)handleResume:(id)sender
+{
+  self.onResume();
+}
+#endif // macOS]
 @end
 
 @implementation RCTPausedInDebuggerOverlayController
 
-#if !TARGET_OS_OSX // [macOS]
-- (UIWindow *)alertWindow
+- (RCTPlatformWindow *)alertWindow // [macOS]
 {
   if (_alertWindow == nil) {
+#if !TARGET_OS_OSX // [macOS]
     _alertWindow = [[UIWindow alloc] initWithWindowScene:RCTKeyWindow().windowScene];
 
     if (_alertWindow) {
       _alertWindow.rootViewController = [UIViewController new];
       _alertWindow.windowLevel = UIWindowLevelAlert + 1;
     }
+#else // [macOS]
+    _alertWindow = [[NSWindow alloc] initWithContentRect:NSMakeRect(0, 0, 100, 100)
+                                               styleMask:NSWindowStyleMaskBorderless | NSWindowStyleMaskFullSizeContentView
+                                                 backing:NSBackingStoreBuffered
+                                                   defer:YES];
+    _alertWindow.backgroundColor = [NSColor clearColor];
+    _alertWindow.opaque = NO;
+#endif // macOS]
   }
 
   return _alertWindow;
 }
-#endif // [macOS]
 
 - (void)showWithMessage:(NSString *)message onResume:(void (^)(void))onResume
 {
   [self hide];
 
   RCTPausedInDebuggerViewController *view = [[RCTPausedInDebuggerViewController alloc] init];
-#if !TARGET_OS_OSX // [macOS]
-  view.modalPresentationStyle = UIModalPresentationOverFullScreen;
   view.message = message;
   view.onResume = onResume;
+  
+#if !TARGET_OS_OSX // [macOS]
+  view.modalPresentationStyle = UIModalPresentationOverFullScreen;
   [self.alertWindow makeKeyAndVisible];
   [self.alertWindow.rootViewController presentViewController:view animated:NO completion:nil];
-#endif // [macOS]
+#else // [macOS]
+  self.alertWindow.contentViewController = view;
+  
+  NSWindow *parentWindow = RCTKeyWindow();
+  if (![[parentWindow sheets] doesContain:self->_alertWindow]) {
+    [parentWindow beginSheet:self.alertWindow completionHandler:^(NSModalResponse returnCode) {
+      [self->_alertWindow orderOut:self];
+    }];
+  }
+#endif // macOS]
 }
 
 - (void)hide
 {
 #if !TARGET_OS_OSX // [macOS]
   [_alertWindow setHidden:YES];
-
   _alertWindow.windowScene = nil;
-
+  _alertWindow = nil;
```

Comment:
Can we keep the newlines, and maybe move `_alertWindow = nil;` to the bottom since it's common for both platforms?